### DISCUSSION TITLE
The response of the "can" method is weird and returning something contradicting to my data.

### DISCUSSION THREAD

**Discussion by nocejapatrick** [spatie/laravel-permission#2084] [Q&A] [answered]
Can anybody explain to me this error?

![image](https://user-images.githubusercontent.com/32543501/166718518-80b661a9-28eb-4ee3-9abf-509cf3b3bcfe.png)

**Comment by jothamhernandez** [comment]
Following up with another line of query

![image](https://user-images.githubusercontent.com/5656719/166718953-397b19c7-d9ff-4097-9d86-993ce1ba1494.png)

This show

> Can anybody explain to me this error?
> 
> ![image](https://user-images.githubusercontent.com/32543501/166718518-80b661a9-28eb-4ee3-9abf-509cf3b3bcfe.png)

**Reply by erikn69** [reply]
do you have two github users?

**Comment by erikn69** [comment, marked as answer]
Looks like the problem is the `guard_name`, take a look on db

**Reply by jothamhernandez** [reply]
My colleague left his account on my computer.

anyway, I still don't have any other guard aside from web.

![image](https://user-images.githubusercontent.com/5656719/166722646-5d7be411-0772-4ed8-93fe-f2e215f92963.png)

**Reply by erikn69** [reply]
The problem is the cache, maybe the cache not has the permissions, try cache refresh before hasPermissionTo, 
`->forgetCachedPermissions()`

**Reply by jothamhernandez** [reply]
thanks for the solution @erikn69,

does the recaching for permissions doesn't included on the optimize artisan command?

**Reply by erikn69** [reply]
It seems not, but check it to confirm it, I'm not the author, I don't know all the functionalities
